http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/leoPubsubMessageSubscriberModels.scala

```diff
@@ -104,13 +116,25 @@ object LeoPubsubCodec {
       "traceId"
     )(CreateRuntimeMessage.apply)
 
+  implicit val deleteRuntimeDecoder: Decoder[DeleteRuntimeMessage] =
+    Decoder.forProduct2("runtimeId", "traceId")(DeleteRuntimeMessage.apply)
+
+  implicit val stopRuntimeDecoder: Decoder[StopRuntimeMessage] =
+    Decoder.forProduct2("runtimeId", "traceId")(StopRuntimeMessage.apply)
+
+  implicit val startRuntimeDecoder: Decoder[StartRuntimeMessage] =
+    Decoder.forProduct2("runtimeId", "traceId")(StartRuntimeMessage.apply)
+
   implicit val leoPubsubMessageDecoder: Decoder[LeoPubsubMessage] = Decoder.instance { message =>
     for {
       messageType <- message.downField("messageType").as[String]
       value <- messageType match {
         case "stopUpdate"         => message.as[StopUpdateMessage]
         case "transitionFinished" => message.as[RuntimeTransitionMessage]
         case "createRuntime"      => message.as[CreateRuntimeMessage]
+        case "deleteRuntime"      => message.as[DeleteRuntimeMessage]
+        case "stopRuntime"        => message.as[StopRuntimeMessage]
+        case "startRuntime"       => message.as[StartRuntimeMessage]
         case other                => Left(DecodingFailure(s"invalid message type: $other", List.empty))
```

Comment:
not in your change, but I think we should get rid of this `other` case...

Reply:
Why? Since it's matching a `String` I thought we should leave it there.. wouldn't we get a `MatchError` otherwise?

Reply:
oh yea...my bad..I thought it's matching an object...maybe worth adding an ADT for `MessageType`...

Reply:
good idea